twoliter/src/lock.rs

```diff
@@ -139,6 +170,111 @@ impl Lock {
         Ok(lock)
     }
 
+    /// Fetches all external kits defined in a Twoliter.lock to the build directory
+    pub(crate) async fn fetch(&self, project: &Project, arch: &str) -> Result<()> {
+        let target_dir = project.project_dir().join("build/external-kits");
+        create_dir_all(&target_dir)
+            .await
+            .context("failed to create external-kits directory")?;
+        let mut kits: Vec<Image> = Vec::new();
+        for image in self.kit.iter() {
+            kits.push(self.extract_kit(&target_dir, image, arch).await?);
+        }
+        let kit_list = serde_json::to_vec(&ExternalKitMetadata {
+            sdk: Image {
+                name: self.sdk.name.clone(),
+                version: self.sdk.version.clone(),
+                vendor: self.sdk.vendor.clone(),
+            },
+            kits,
+        })
+        .context("failed to serialize external kit metadata")?;
+        write(
+            target_dir.join("external-kit-metadata.json"),
+            kit_list.as_slice(),
+        )
+        .await
+        .context("failed to write external kit metadata")?;
+
+        Ok(())
+    }
+
+    async fn extract_kit<P>(&self, path: P, image: &LockedImage, arch: &str) -> Result<Image>
+    where
+        P: AsRef<Path>,
+    {
+        let tmp_dir =
+            tempdir().context("failed to create a temporary directory for kit extraction")?;
+        let vendor = image.vendor.clone();
+        let name = image.name.clone();
+        let docker_arch = match arch {
+            "x86_64" => Ok("amd64"),
+            "aarch64" => Ok("arm64"),
+            _ => Err(anyhow!("unsupported architecture '{}'", arch)),
+        }?;
+        let target_path = path.as_ref().join(format!("{vendor}/{name}/{arch}"));
+        create_dir_all(&target_path).await?;
+
+        // First we want to get the correct digest for our architecture
+        let manifest_bytes = docker!(
+            ["manifest", "inspect", image.source.as_str()],
+            format!("failed to find a kit {}", image.to_string())
+        );
+        let manifest_list: ManifestListView = serde_json::from_slice(manifest_bytes.as_slice())
+            .context("failed to deserialize manifest list")?;
+        let manifest = manifest_list
+            .manifests
+            .iter()
+            .find(|x| x.platform.as_ref().unwrap().architecture == docker_arch)
+            .context(format!(
+                "could not find kit image for architecture '{}' at {}",
+                arch, image.source
+            ))?;
```

Comment:
It seems like a few smaller functions could be factored out to improve readability:

```suggestion
        let manifest = self.get_manifest(image.source.as_str(), image, docker_arch)?;
```